src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,468 @@
+                         closedCaption: false };
+        newTextTracks.push({ track,
+                             origTrack: textTrack });
+      }
+      return newTextTracks;
+    };
+
+    this._textTracks = [];
+    if (mediaElement.textTracks != null) {
+      this._textTracks = createTextTracks();
+      mediaElement.textTracks.onaddtrack = () => {
+        const newTextTracks = createTextTracks();
+        if (areTrackArraysDifferent(this._textTracks, newTextTracks)) {
+          this._textTracks = newTextTracks;
+          this._setInitialTextTrack();
+          this.trigger("availableTracksChange", "text");
+        }
+      };
+      (mediaElement.textTracks as ICustomTextTrackList).onremovetrack = () => {
+        const newTextTracks = createTextTracks();
+        if (areTrackArraysDifferent(this._textTracks, newTextTracks)) {
+          this._textTracks = newTextTracks;
+          this._setInitialTextTrack();
+          this.trigger("availableTracksChange", "text");
+        }
+      };
+    }
+
+    // Create video tracks from native video tracks.
+    const createVideoTracks = (): Array<{ track: { id: string;
+                                                   representations: []; };
+                                          origTrack: VideoTrack; }> => {
+      const newVideoTracks = [];
+      const { videoTracks } = mediaElement;
+      const languagesOccurences: Partial<Record<string, number>> = {};
+      for (let i = 0; i < videoTracks.length; i++) {
+        const videoTrack = videoTracks[i];
+        const language = videoTrack.language === "" ? "nolang" :
+                                                      videoTrack.language;
+        const occurences = languagesOccurences[language] ?? 1;
+        const id = "gen_video_" +
+                   language +
+                   "_" +
+                   occurences.toString();
+        languagesOccurences[language] = occurences + 1;
+        newVideoTracks.push({ track: { id,
+                                       representations: [] as [] },
+                              origTrack: videoTrack });
+      }
+      return newVideoTracks;
+    };
+
+    this._videoTracks = [];
+    if (mediaElement.videoTracks != null) {
+      this._videoTracks = createVideoTracks();
+      mediaElement.videoTracks.onaddtrack = () => {
+        const newVideoTracks = createVideoTracks();
+        if (areTrackArraysDifferent(this._videoTracks, newVideoTracks)) {
+          this._videoTracks = newVideoTracks;
+          this.trigger("availableTracksChange", "video");
+        }
+      };
+      mediaElement.videoTracks.onremovetrack = () => {
+        const newVideoTracks = createVideoTracks();
+        if (areTrackArraysDifferent(this._videoTracks, newVideoTracks)) {
+          this._videoTracks = newVideoTracks;
+          this.trigger("availableTracksChange", "video");
+        }
+      };
+    }
+
+    this._onTrackChange(mediaElement);
```

Comment:
So this is not called when the track change but a single time at the end of the constructor?

Reply:
Done